compiler/rustc_mir/src/transform/check_unsafety.rs

```diff
@@ -190,7 +190,7 @@ impl<'a, 'tcx> Visitor<'tcx> for UnsafetyChecker<'a, 'tcx> {
             }
         }
 
-        for (i, elem) in place.projection.iter().enumerate() {
+        for (i, _elem) in place.projection.iter().enumerate() {
             let proj_base = &place.projection[..i];
             if context.is_borrow() {
```

Comment:
Pre-existing but doesn't this have the same bug around derefs? i.e., `&(*foo.bar).baz` will be considered unsafe if `bar` is misaligned, but that's not necessary, right?

Reply:
I agree, `*foo.bar` will *always* read the `foo.bar` value correctly, then use it as the base for `&_.baz` borrow.

Reply:
In fact, `util::is_disaligned` already walks down the projections (until a `Deref`). So the check inside the loop here is simply unnecessary.

Reply:
But I cannot actually make thus bug surface? The following code works fine (no warning about packed fields):
```rust
#![warn(unaligned_references)]
struct S(u32);

#[repr(packed)]
struct P(&'static S);

fn main() {
    let p = P(&(S(0)));
    let x = &(*p.0).0;
}
```

Reply:
Ah right, the check actually does not consider the loop variable! It just re-checks the top-level place each time around the loop. That's silly, but doesn't behave incorrectly.